test/fixtures/azure-fixture/src/main/java/fixture/azure/MockAzureBlobStore.java

```diff
@@ -0,0 +1,483 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package fixture.azure;
+
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.bytes.CompositeBytesReference;
+import org.elasticsearch.common.util.Maps;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.rest.RestStatus;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+public class MockAzureBlobStore {
+
+    private static final Logger logger = LogManager.getLogger(MockAzureBlobStore.class);
+    private static final BytesReference EMPTY = new BytesArray(new byte[] {});
```

Comment:
I didn't spot that it was available, but when I switched over to it, it caused some issues. I used `contents == EMPTY` to determine if a blob was uncommitted, and when I switch to using `BlobArray.EMPTY` that breaks (I assume because the constant is used elsewhere to represent the body of a zero length request?

I renamed the private constant to `UNCOMMITTED` in 107d45d5eead51b7e6b11f269664d8fff0137043 to make it clearer. It's a bit hacky but better than keeping additional state to signify that IMO.

Reply:
We could probably leave the contents as null for uncommitted now that I've fixed an uncommitted blob to return 404 instead of empty bytes.